gitbook-docs/security.md

```diff
@@ -8,8 +8,19 @@ with authentication and access control mechanisms in place.
 
 ## Authentication
 
-Authentication for Signal K REST and WebSockets connections is based on HTTP cookies or tokens carried in the HTTP
-header.
+Authentication for Signal K connections is based on a token carried in the message, or in a cookie or tokens carried in the HTTP
+`Authorization` header for a HTTP request. The tokens can be of any type.
+
+__Note__: a signalk server should never simply echo or redistribute a message received without removing or replacing the token. 
+That would result in A's token being sent to B, which allows the B to impersonate A.  
+
+There are 3  authentication actions:
+
+* authenticate - login and obtain a token
+* logout - invalidate a token
+* validate - validate a token with auto-renewal if valid. 
+
+All 3 actions can be done via HTTP REST semantics for web based clients, or by sending Signal K messages for others.
```

Comment:
Clients may be sensors and still use http, no web involved, and there isn't anything particularly REST about the mechanism. I suggest _All 3 actions can be done via HTTP requests or by sending Signal K messages for non HTTP clients._

Reply:
changed